.github/workflows/connector_integration_test_single_dagger.yml

```diff
@@ -27,30 +39,57 @@ jobs:
           aws-secret-access-key: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
           github-token: ${{ env.PAT }}
           ec2-instance-type: "c5.2xlarge" # https://aws.amazon.com/ec2/instance-types/
-  launch_integration_tests:
+  launch_connector_tests:
     timeout-minutes: 120
-    runs-on: ${{ needs.start-connectors-integration-test-runner.outputs.label }} # run the job on the newly created runner
+    needs: start-runner
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runner
+    env:
+      GCP_GSM_CREDENTIALS: ${{ secrets.GCP_GSM_CREDENTIALS }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.STATUS_API_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.STATUS_API_AWS_SECRET_ACCESS_KEY }}
+      AWS_DEFAULT_REGION: "us-east-2"
+      TEST_REPORTS_BUCKET_NAME: "airbyte-connector-build-status"
     steps:
-      - name: Checkout Airbyte
+      - name: Checkout master
         uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repo }}
+          ref: master
+      - name: Checkout current branch
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repo }}
+          ref: ${{ github.event.inputs.gitref }}
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
       - name: Install Python 3.10
         uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-      - name: Upgrade pip
-        run: pip install --upgrade pip
       - name: Install ci-connector-ops package
         run: pip install --quiet -e ./tools/ci_connector_ops
-  stop-connectors-integration-test-runner:
+      - name: Run connectors-ci for specific connectors
+        if: "${{ github.event.inputs.connectors_under_test != '' }}"
+        run: connectors-ci --is-local=False test-connectors ${{ github.event.inputs.connectors_under_test }}
+        env:
+          CI_GIT_BRANCH: ${{ steps.extract_branch.outputs.branch }}
+      - name: Run connectors-ci for modified connectors
+        if: github.event_name == 'pull_request' || github.event_name == 'push' || github.event.inputs.test_modified_connectors == 'true'
```

Comment:
A bit wordy, but feels necessary due to the event input.

Also, push is a bit greedy here. I think you should only run on push to master instead of just push so that we aren't spinning up a workflow every time that someone pushes an arbitrary branch that isn't part of a pull request. You can do that by modifying the "on:" push clause to only include master branch above

Reply:
But how would we get the guarantee that all the connectors changed in a PR are passing tests before merge? 
Do you mean we should avoid the on-push for the moment until we get the infra with remote cache ready?

Reply:
> But how would we get the guarantee that all the connectors changed in a PR are passing tests before merge?

The pull_request event handles that. Basically it will run every time you push to a branch, as long as there is an open pull request for that branch. 

restricting push to only run on master basically prevents invocation of the tests when you push to a non-master branch with no pull request open on it. But it would run as soon as you opened the pull request and also to subsequent pushes to that branch, so long as the PR is open.